zetaclient/bitcoin/bitcoin_client.go

```diff
@@ -690,7 +690,7 @@ func (ob *BTCChainClient) GetInboundVoteMessageFromBtcEvent(inTx *BTCInTxEvnet)
 		inTx.FromAddress,
 		ob.chain.ChainId,
 		inTx.FromAddress,
-		inTx.FromAddress,
+		inTx.ToAddress,
```

Comment:
Thanks for pointing this out; I believe the address is fetched from the `message` field. 
This value is not used; adding the ToAddress would make logical sense. 

However, I can replace this with logic like 
```
toAddress := inTx.ToAddress,
address := parseAddressFromMessage(message)
if address.Valid() {
  toAddress = address
}

```

It would be great to get your thoughts @ws4charlie

Reply:
This `ToAddress` in `BTCInTxEvnet` is the TSS bitcoin address. I think we can just leave it untouched as I saw the same logic in evm inbound tx (gas token) observation, meaning sending gas token to the sender itself. Zetaclient doesn't interpret the meaning of the memo (address, call data, e.g.). Zetacore should be able to know the correct value to be set in a cctx's true `receiver` in zEVM.
For example: http://100.71.167.102:1317/zeta-chain/crosschain/in_tx_hash_to_cctx_data/847139aa65aa4a5ee896375951cbf7417cfc8a4d6f277ec11f40cd87319f04aa
The receiver is now equal to sender (I think is not semantically correct): `bc1q68kxnq52ahz5vd6c8czevsawu0ux9nfrzzrh6e`
The memo in hex is: `6a1467ed0bcc4e1256bc2ce87d22e190d63a120114bf`
The decoded receiver is: `0xeBb79Dd1B71ce1ED76E7a6Dcd9C7bCEDddb67B5f`, I was looking at ZetaScan: https://explorer.zetachain.com/cc/tx/0x9114a8d7e7441cee7c38232dc1ce41de61beb08c12ab988662703b64248ba304

![image](https://github.com/zeta-chain/node/assets/34498985/d58d8c57-2d5b-47f9-a8a7-9489bf4862cd)

Reply:
861b635a0eac3ff51a6ae22f1c3f2124fab12f1a